pkg/controller/node/node_controller.go

```diff
@@ -197,23 +200,28 @@ func inSlice(str string, s []string) bool {
 	return false
 }
 
-func filterPortsByNodeAddresses(nsxClients *NsxClients, ports *[]nsxtmgr.LogicalPort, nodeAddresses []string) ([]*nsxtmgr.LogicalPort, error) {
-	var filteredPorts []*nsxtmgr.LogicalPort
+func filterPortsByNodeAddresses(nsxClients *NsxClients, ports *[]model.SegmentPort, nodeAddresses []string) ([]*model.SegmentPort, error) {
+	var filteredPorts []*model.SegmentPort
 	log.Info(fmt.Sprintf("Found %d ports for node, checking addresses %v", len(*ports), nodeAddresses))
-	nsxClient := nsxClients.ManagerClient
+	connector := nsxClients.PolicyConnector
+	segPortStateClient := segPortSDK.NewStateClient(connector)
 	for _, port := range *ports {
-		logicalPort, _, err := nsxClient.LogicalSwitchingApi.GetLogicalPortState(nsxClient.Context, port.Id)
+		segId, err := operatortypes.ExtractSegmentIdFromPath(*port.ParentPath)
 		if err != nil {
 			return filteredPorts, err
 		}
-		if len(logicalPort.RealizedBindings) == 0 {
+		portState, err := segPortStateClient.Get(segId, *port.Id, nil, nil)
+		if err != nil {
+			return filteredPorts, err
+		}
```

Comment:
For this check and the one at line 210 I'm thinking that perhaps we want to do thing a little differently:
Instead of returning at the first error, keep iterating over ports, and then return slices for FilteredPorts and errors.

The thing is that we want to find "the" port with the right IP, so even if we encounter issues while fetching or parsing state for other ports, we might still be able to find the port we are looking for.

Reply:
what we can do alternatively is finish the loop collecting errors. If there are any ports which couldn't be read from NSX, we return 1 error providing the info of all erroneous ports

Reply:
sounds good!